metrics/otelgrpc/otelgrpc.go

```diff
@@ -0,0 +1,162 @@
+package otelgrpc
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/goto/meteor/utils"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+)
+
+type UnaryParams struct {
+	Start  time.Time
+	Method string
+	Req    any
+	Res    any
+	Err    error
+}
+
+type Monitor struct {
+	duration     metric.Int64Histogram
+	requestSize  metric.Int64Histogram
+	responseSize metric.Int64Histogram
+	attributes   []attribute.KeyValue
+}
+
+func NewOtelGRPCMonitor(hostName string) Monitor {
+	meter := otel.Meter("github.com/goto/meteor/metrics/otelgrpc")
+
+	duration, err := meter.Int64Histogram("rpc.client.duration", metric.WithUnit("ms"))
+	if err != nil {
+		handleOtelErr(err)
+	}
+
+	requestSize, err := meter.Int64Histogram("rpc.client.request.size", metric.WithUnit("By"))
+	if err != nil {
+		handleOtelErr(err)
+	}
+
+	responseSize, err := meter.Int64Histogram("rpc.client.response.size", metric.WithUnit("By"))
+	if err != nil {
+		handleOtelErr(err)
+	}
+
+	return Monitor{
+		duration:     duration,
+		requestSize:  requestSize,
+		responseSize: responseSize,
+		attributes: []attribute.KeyValue{
+			semconv.RPCSystemGRPC,
+			attribute.String("server.address", hostName),
+		},
+	}
+}
+
+func getProtoSize(p any) (int, error) {
+	if p == nil {
+		return 0, nil
+	}
+
+	data, err := proto.Marshal(p.(proto.Message))
```

Comment:
use [`proto.Size`](https://pkg.go.dev/github.com/golang/protobuf/proto#Size) ?

Reply:
yes it's another approach as well. https://github.com/goto/meteor/pull/30#discussion_r1282710038